lib/vrl/lookup/src/lookup_v2/mod.rs

```diff
@@ -1,3 +1,54 @@
+//! This crate contains all of the logic for lookup paths.
+//!
+//! # Traits
+//! There are 2 main traits that define a path.
+//! - [ValuePath] is a path that points inside of a `Value`
+//! - [TargetPath] is a path that points to a `target`. A `target` in VRL refers to
+//! the external data being processed by a VRL script. A `target` has 2 main sections that can be
+//! pointed to, `event` and `metadata`.  [TargetPath::prefix] identifies the section, and
+//! [TargetPath::value_path] is a [ValuePath] pointing into that section.
+//!
+//! Note that for performance reasons, since [ValuePath] and [TargetPath] require [Clone], these
+//! traits are only implemented on types that are cheap to clone (usually references). That means
+//! when passing in a value (e.g. [OwnedValuePath]) into a function that requires `impl ValuePath`,
+//! you will generally need to pass it in as a reference.
+//!
+//! # Owned Paths
+//! [OwnedValuePath] and [OwnedTargetPath] are fully parsed paths. These contain heap-allocated
+//! segments.
+//!
+//! Owned paths should generally be preferred over other types (such as strings) whenever feasible
```

Comment:
For efficiency? Or ...

And,

```suggestion
//! Owned paths should generally be preferred over other types (such as strings) whenever feasible.
```